apps/src/netsim/NetSimLobby.js

```diff
@@ -48,415 +61,324 @@ var NetSimLobby = module.exports = function (levelConfig, connection, user,
-    this.refreshLobbyList_(lobbyData);
-  }.bind(this));
 };
 
 /**
- * Reload the lobby listing of nodes.
- * @param {!Array.<NetSimClientNode>} lobbyData
+ * Handler for clicking the "Cancel" button to stop trying to connect to
+ * another client.
  * @private
  */
-NetSimLobby.prototype.refreshLobbyList_ = function (lobbyData) {
-  this.lobbyList_.empty();
-
-  // TODO: Filter based on level configuration
-  var filteredLobbyData = lobbyData.filter(function (simNode) {
-    var showClients = this.levelConfig_.showClientsInLobby;
-    var showRouters = this.levelConfig_.showRoutersInLobby;
-    var nodeType = simNode.getNodeType();
-    return (nodeType === NodeType.CLIENT && showClients) ||
-        (nodeType === NodeType.ROUTER && showRouters);
-  }.bind(this));
-
-  filteredLobbyData.sort(function (a, b) {
-    // TODO (bbuchanan): Make this sort localization-friendly.
-    if (a.getDisplayName() > b.getDisplayName()) {
-      return 1;
-    }
-    return -1;
-  });
-
-  this.selectedListItem_ = undefined;
-  filteredLobbyData.forEach(function (simNode) {
-    var item = $('<li>').html(
-        simNode.getDisplayName() + ' : ' +
-        simNode.getStatus() + ' ' +
-        simNode.getStatusDetail());
-
-    // Style rows by row type.
-    if (simNode.getNodeType() === NodeType.ROUTER) {
-      item.addClass('router-row');
-    } else {
-      item.addClass('user-row');
-      if (simNode.entityID === this.connection_.myNode.entityID) {
-        item.addClass('own-row');
-      }
-    }
-
-    // Preserve selected item across refresh.
-    if (simNode.entityID === this.selectedID_) {
-      item.addClass('selected-row');
-      this.selectedListItem_ = item;
-    }
-
-    item.click(this.onRowClick_.bind(this, item, simNode));
-    item.appendTo(this.lobbyList_);
-  }.bind(this));
+NetSimLobby.prototype.onCancelButtonClick_ = function () {
+  this.netsim_.disconnectFromRemote();
+};
 
-  this.onSelectionChange();
+/**
+ * Called whenever a change is detected in the nodes table - which should
+ * trigger a refresh of the lobby listing
+ * @param {!Array} rows
+ * @private
+ */
+NetSimLobby.prototype.onNodeTableChange_ = function (rows) {
+  this.nodesOnShard_ = netsimNodeFactory.nodesFromRows(this.shard_, rows);
+  this.render();
 };
 
 /**
- * @param {*} connectionTarget - Lobby row for clicked item
+ * Called whenever a change is detected in the wires table.
+ * @param {!Array} rows
  * @private
  */
-NetSimLobby.prototype.onRowClick_ = function (listItem, connectionTarget) {
-  // Can't select user rows (for now)
-  if (NodeType.CLIENT === connectionTarget.getNodeType()) {
-    return;
-  }
+NetSimLobby.prototype.onWireTableChange_ = function (rows) {
+  // Update the collection of nodes with connections pointing toward us.
+  this.nodesRequestingnetsim_ = rows.filter(function (wireRow) {
+    return wireRow.remoteNodeID === this.myNode_.entityID;
+  }.bind(this)).map(function (wireRow) {
```

Comment:
I could see either way, just curious—any reason / preference for using `.bind(this)` over using `this` as the second param to `filter`, like `.filter(function(){}, this)`?